crates/engine_provider/src/auth_provider.rs

```diff
@@ -209,17 +214,33 @@ where
         payload: ExecutionPayload,
         parent_beacon_block_root: Option<B256>,
         versioned_hashes: Vec<B256>,
+        execution_requests: Option<Vec<Bytes>>,
     ) -> TransportResult<EngineApiMessageVersion> {
         match payload {
             ExecutionPayload::V3(payload) => {
-                // We expect the caller
-                let parent_beacon_block_root = parent_beacon_block_root
-                    .expect("parent_beacon_block_root is required for V3 payloads");
-                self.inner
-                    .new_payload_v3_wait(payload, versioned_hashes, parent_beacon_block_root)
-                    .await?;
-
-                Ok(EngineApiMessageVersion::V3)
+                if payload.payload_inner.timestamp() >= MAINNET_PRAGUE_TIMESTAMP {
```

Comment:
this isn't necessarily reflected on all target chains, especially devnets, so it's not actually a reliable way to detect the hard-fork of a chain